### ISSUE TITLE
lsp-fsharp prints "No ident at this location" error

### ISSUE THREAD

**Issue by fehmud** [emacs-lsp/lsp-mode#2000]
Hello everybody and thank you for this package.

I am experiencing a problem with `lsp-fsharp` that should be easy to reproduce, therefore I am skipping the full bug report for now.

The problem is that when point is on an operator - like `|` - `lsp-mode` prints the message:

    LSP :: Error from the Language Server: No ident at this location (Internal Error)

**Comment by yyoncho** [comment]
Do you have lsp-ui enabled? I think that this "error" is hidden in lsp-mode.

**Comment by fehmud** [comment]
@yyoncho 

Thanks for your suggestion. I have disabled `lsp-ui`, but the warning persists.

As a workaround until I will investigate this better, I have redefined `lsp--create-default-error-handler` to generate a handler that bypasses that warning.

**Comment by yyoncho** [comment]
It will be great if you can provide the location of the code that triggers the error(or someone using FSharp). Note that we have 40+ languages and even configuring each one of them + sample project is time-consuming if you are not using that language.

**Comment by fehmud** [comment]
I have put a conditional breakpoint on `log--warn`, and the warning seems coming from a "textDocument/documentHighlight" response (the earliest visible function call on the stack is `lsp--parser-on-message`).

When the token under point is not an identier, then `lsp-fsharp` may:
1. warn of "No ident at this location" (e.g. on `|`, `10`);
2. warn of "No symbol information found" (e.g. on `=`, `_`);
3. like #2 and then echo the symbol in the minibuffer (e.g. on `let`, `->`).

**Comment by s-kostyaev** [comment]
Is there any examples for that. I can't understand how to use it.

**Comment by yyoncho** [comment]
@s-kostyaev no, I mean that we can change lsp-mode to allow that pattern. E. g. `(defcustom lsp-default-error-handler ...` which then can be passed at https://github.com/yyoncho/lsp-mode/blob/posframe-better/lsp-mode.el#L3258

**Comment by yyoncho** [comment]
or it can be `lsp-default-create-error-handler-fn` to allow passing the method name which can be convenient for filtering the errors.

**Comment by s-kostyaev** [comment]
I think about variant where we can pass it into `make-lsp-client` for default error handler per lsp-client.

**Comment by s-kostyaev** [comment]
Or another option: add something like `:ignore-errors` into `make-lsp-client` and filter it inside `lsp--create-default-error-handler`.

**Comment by yyoncho** [comment]
That option is harder to implement and having it in lsp-client will make customization more difficult.

**Comment by s-kostyaev** [comment]
We have already implemented `:ignore-messages` inside `make-lsp-client`. I think create similar filtering inside default-error-handler not that hard. I can try implement it myself.

**Comment by s-kostyaev** [comment]
For now workaround is something like:
```elisp
  (setq my-lsp-skip-errors
	'("Index was outside the bounds of the array."
	  "No symbol information found"
	  "No ident at this location"))
  (setq lsp-default-create-error-handler-fnlsp-default-create-error-handler-fn
	(lambda (method)
	  (lambda (error)
	    (when (not (seq-find (lambda (s)
				   (string= s (lsp-get error :message)))
				 my-lsp-skip-errors))
	      (lsp--warn "%s" (or (lsp--error-string error)
				  (format "%s Request has failed" method)))))))
```
@yyoncho Maybe we also can `(setq-local lsp-default-create-error-handler-fn ...` inside lsp-client?

**Comment by s-kostyaev** [comment]
upstream bug https://github.com/fsharp/FsAutoComplete/issues/833
